### ISSUE TITLE
Feature Request: Add an "Editorial" Feature for Problem Solutions

### ISSUE THREAD

**Issue by Mateuszciesla1** [sio2project/oioioi#424]
It would be beneficial to implement a feature allowing users to access editorial solutions for problems. This feature would introduce a new option, "Editorial", alongside the existing options:

- Problem Statement
- Downloads
- Submissions
- Submit
- Related Portal Pages

The "Editorial" tab would contain a detailed explanation of the problem's solution, aiding users in understanding how to approach and solve the task. An additional configuration could be introduced for contest administrators. This setting would allow them to unlock editorial solutions for participants after the contest ends.

**Comment by segir187** [comment]
Good idea! As of now, I am too inexperienced to implement such a feature and preoccupied with a different one, but I like it. It should probably be invisible for tasks without editorials, since in huge task databases, such as Szkopuł, almost no tasks have Editorials and showing an "Editorial" option opening sites displaying messages like "No editorial :(" would set users up for a lot of disappointment.

**Comment by segir187** [comment]
Sorry, closed accidentally!

**Comment by kama-pa-go** [comment]
Almost half of this issue is already done and can be found in #525 PR (the "base editorial implementation"  part).
